deepspeed/moe/sharded_moe.py

```diff
@@ -366,7 +366,7 @@ def top2gating(logits: Tensor,
     combine_weights = combine1_sec + combine2_sec
     dispatch_mask = combine_weights.bool()
 
-    return l_aux, combine_weights, dispatch_mask, exp_counts.detach().to('cpu')
+    return l_aux, combine_weights, dispatch_mask, exp_counts
```

Comment:
Currently `exp_counts` is unused at `forward()` of any of MoE classes, right?

Reply:
> Currently `exp_counts` is unused at `forward()` of any of MoE classes, right?

Yes, I have test it in Megatron-deepspeed and find that `exp_counts` is unused at `forward()` of any of MoE classes.

Reply:
I have test these changes on my testing GPU platform, and work fine, no error and the loss keeps same to the original way.